lib/routes/account.js

```diff
@@ -2169,6 +2170,22 @@ module.exports = (
             })
         }
 
+        function deleteSecondaryEmailIfUnverified() {
+          return db.getSecondaryEmail(email)
+            .then((secondaryEmailRecord) => {
+              // Only delete secondary email if it is unverified and does not belong
+              // to the current user.
+              if (! secondaryEmailRecord.isVerified && secondaryEmailRecord.uid !== uid.toString('hex')) {
```

Comment:
*Adding to mental backlog of papercuts to fix*: all the `toString('hex')` we have to do when comparing. Agh, I just reallized that in JavaScript, custom classes can't override the equal operator, so it'd still be method call. Oh well, it could `Uid.equals`...